src/lib/testing/utils.ts

```diff
@@ -37,13 +44,51 @@ export const testWithDb = <T>(
                 .join(', ')} RESTART IDENTITY CASCADE;`;
             await db.execute(sql.raw(query));
 
-            const generated = generator ? await generator(db) : undefined;
-
-            const newArgs = testArgs as TestArgs<T>;
+            const newArgs = testArgs as TestArgs;
             newArgs.db = db;
-            newArgs.generated = generated;
 
-            await testFunc(newArgs);
+            if (generator) {
+                await testFunc(newArgs, mustGenerate(await generator(db)));
+            } else {
+                await (testFunc as (args: TestArgs) => Promise<void>)(newArgs);
+            }
         });
     });
+}
+
+// mimics an iterator, but not an actual iterator so we don't need to do `.next().value`
+export const sequentialDates = (
+    start: Date | number = new Date(2020, 0),
+    seconds_diff: number = 60 * 60 * 24
+) => {
+    let curr = +start;
+    const r = {
+        next: () => {
+            const ret = curr;
+            curr += seconds_diff;
+            return new Date(ret);
+        },
+        take: (n: number) => {
+            return Array(n).fill(undefined).map(r.next);
+        },
+    };
+    return r;
+};
+
+export const assert_nonnull = <T>(t: T): t is NonNullable<T> => {
+    if (t === null) {
+        throw new Error('Assertion failed.  t === null');
+    }
+    return true;
+};
+
+export const create_users = async (
```

Comment:
nit: snake_case should be camelCase here.

Reply:
Fixed, but snake_case looks better :)